protocol-jackson/src/main/java/io/camunda/zeebe/protocol/jackson/record/ZeebeStyle.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.jackson.record;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Style.BuilderVisibility;
+import org.immutables.value.Value.Style.ImplementationVisibility;
+import org.immutables.value.Value.Style.ValidationMethod;
+
+@Target({ElementType.PACKAGE, ElementType.TYPE})
+@Retention(RetentionPolicy.CLASS)
+@Value.Style(
```

Comment:
Happy to add comments to the settings and why I picked them, however they are very well documented so it shouldn't be too hard to figure it out. Let me know what you think.

Reply:
I don't think all options require explaining, but some definitely have been chosen due to some Zeebe intricacies. 

🔧  Please  add some explanation to: 

```
builderVisibility = BuilderVisibility.PUBLIC, // let's define why its public
implementationNestedInBuilder = true, // discussed above, could use a summary explanation here
defaultAsDefault = true, // unclear to me why we even need this
deepImmutablesDetection = true, // does quite a lot, so why exactly do we need it?
```